### PR TITLE
GEODE-6205: Set disk-dir in gfsh either absolute or relative based on…

### PR DESCRIPTION
**Pull request by boglesby**
… input directory

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/DiskStoreCommandsDUnitTest.java [hunk 1]
```diff
@@ -65,10 +71,10 @@
   public ClusterStartupRule rule = new ClusterStartupRule();
 
   @Rule
-  public GfshCommandRule gfsh = new GfshCommandRule();
+  public transient GfshCommandRule gfsh = new GfshCommandRule();
```
[the remark this answers is not in the document]
Comment (boglesby) [reply geode-core/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/DiskStoreCommandsDUnitTest.java]: Yes, exactly. And I changed DiskStoreCommandsDUnitTest to be Serializable so it serializes when the lambdas are executed. Otherwise an exception like this is thrown:
```
org.apache.geode.test.dunit.RMIException: While invoking org.apache.geode.management.internal.cli.commands.DiskStoreCommandsDUnitTest$$Lambda$53/48305285.run in VM 1 running on Host localhost with 4 VMs
	at org.apache.geode.test.dunit.VM.execute(VM.java:537)
	at org.apache.geode.test.dunit.VM.invoke(VM.java:425)
	at org.apache.geode.test.dunit.VM.invoke(VM.java:401)
	at org.apache.geode.test.dunit.VM.invoke(VM.java:360)
	at org.apache.geode.test.junit.rules.VMProvider.invoke(VMProvider.java:85)
	at org.apache.geode.management.internal.cli.commands.DiskStoreCommandsDUnitTest.validateDiskStoreDiskDirectoryPath(DiskStoreCommandsDUnitTest.java:488)
Caused by: java.rmi.MarshalException: error marshalling arguments; nested exception is: 
	java.io.NotSerializableException: org.apache.geode.management.internal.cli.commands.DiskStoreCommandsDUnitTest
	at sun.rmi.server.UnicastRef.invoke(UnicastRef.java:158)
	at java.rmi.server.RemoteObjectInvocationHandler.invokeRemoteMethod(RemoteObjectInvocationHandler.java:227)
	at java.rmi.server.RemoteObjectInvocationHandler.invoke(RemoteObjectInvocationHandler.java:179)
	at com.sun.proxy.$Proxy32.executeMethodOnObject(Unknown Source)
	at org.apache.geode.test.dunit.VM.execute(VM.java:534)
```

**Comment by boglesby** [comment]
@Petahhh: Yes, thats right. This is changing what is persisted in cluster configuration. Without the change, this command:
```
create disk-store --name=DISKSTORE --dir=/absolute/path/to/myDiskDir
```
Saved the relative `disk-dir` to cluster configuration:
```
<disk-store name="DISKSTORE" ...>
  <disk-dirs>
    <disk-dir>myDiskDir</disk-dir>
  </disk-dirs>
</disk-store>
```
When a new server was started, it created a DiskStore in the working directory instead of the absolute directory.

With the change, this command:
```
create disk-store --name=DISKSTORE --dir=/absolute/path/to/myDiskDir
```
Saves the absolute `disk-dir` to cluster configuration:
```
<disk-store name="DISKSTORE" ...>
  <disk-dirs>
    <disk-dir>/absolute/path/to/myDiskDir</disk-dir>
  </disk-dirs>
</disk-store>
```
And this command:
```
create disk-store --name=DISKSTORE --dir=myDiskDir
```
Saves the relative `disk-dir` to cluster configuration:
```
<disk-store name="DISKSTORE" ...>
  <disk-dirs>
      <disk-dir>myDiskDir</disk-dir>
  </disk-dirs>
</disk-store>
```
In both cases, the server gets the correct disk directory.

Basically whatever is typed as the value of the `dir` in the command is saved to cluster configuration.
